### PR TITLE
PHOENIX-6152 Add phoenix-shaded-protobuf to phoenix-thirdpart.

### PR DESCRIPTION
**Pull request by richardantal**

### CODE REVIEW

File: phoenix-shaded-protobuf/pom.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  <parent>
+    <groupId>org.apache.phoenix.thirdparty</groupId>
+    <artifactId>phoenix-thirdparty</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>..</relativePath>
+  </parent>
+  <artifactId>phoenix-shaded-protobuf</artifactId>
+  <name>Apache Phoenix Relocated (Shaded) Protobuf</name>
+  <description>
+    Pulls down protobuf, compiles, and then relocates/shades.
```
Comment (stoty) [inline phoenix-shaded-protobuf/pom.xml]: This comment is still not true

File: phoenix-shaded-protobuf/pom.xml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  </description>
+  <build>
+    <plugins>
+      <plugin>
+        <!--Clean needs to purge src/main/java since this is where
+             the unpack of protobuf is overlaid. Do it for usual
+             clean goal. We use src/main/java instead of dir
+             under 'target' because the jar plugin is dumb, hard to
+             make it source from other then src/main/java.-->
+        <artifactId>maven-clean-plugin</artifactId>
```
Comment (stoty) [inline phoenix-shaded-protobuf/pom.xml:39]: The clean plugin settings can also be take from guava.

**Review by stoty — CHANGES_REQUESTED** [review]
Also, please create and link a PR for the corresponding changes in phoenix-core, so that we can test them together.

**Comment by richardantal** [comment]
Here is the Phoenix [PR](https://github.com/apache/phoenix/pull/894).

**Pull request closed without merging**